habitica.el

```diff
@@ -591,31 +737,93 @@ NEW-TAG is the new name to give to the tag."
       (replace-match (concat ":" new-tag ":"))))
   (org-align-all-tags))
 
+(defun habitica--goto-task (id)
+  "Goto habitica task according to `ID'"
+  (goto-char (org-find-property "HABITICA_ID" id)))
+
+(defun habitica-buffer-p (&optional buf)
+  "Judge if `buf' is the habitica buffer"
+  (let ((buf (or buf (buffer-name))))
+    (with-current-buffer buf
+      (string= (buffer-name) habitica-buffer-name))))
+
+(defun habitica-task-done-up ()
+  "Mark habitic task DONE makes the score up"
+  (unless habitica-triggered-state-change-p
+    (ignore-errors
+      (let ((habitica-id (org-element-property :HABITICA_ID (org-element-at-point)))
+            (habitica-triggered-state-change-p t)
+            new-score)
+        (while (and (null habitica-id)
+                    (org-up-heading-safe))
+          (setq habitica-id (org-element-property :HABITICA_ID (org-element-at-point))))
+        (when (and habitica-id
+                   org-state
+                   (string= org-state "DONE"))
+          ;; (habitica-api-score-task habitica-id "up")
+          (with-current-buffer habitica-buffer-name
+            (save-excursion
+              (habitica--goto-task habitica-id)
+              (habitica-up-task)
+              (habitica--goto-task habitica-id)
+              (setq new-score (org-element-property :HABITICA_VALUE (org-element-at-point)))))
+          ;; update the HABITICA_VALUE in origin org files
+          (save-excursion
+            (org-back-to-heading)
+            (org-set-property "HABITICA_VALUE" new-score))
+          )))))
+
+
 ;;;; Interactive
+
+(defun habitica-cron ()
+  "Runs cron"
+  (interactive)
+  (habitica-api-cron))
+
+(defun habitica-insert-selected-task (&optional tasks)
```

Comment:
that's super useful. thank you for adding. 
I just wish it could insert it in the buffer at the correct location rather than under the cursor.

Reply:
What is the correct location? Is there a way to get the correct location?

Reply:
No not really...

Reply:
I'm new to Emacs, but I recall seeing a keybinding in evil-org docs that suggest they have a way to try to guess what to insert on a new line (I believe it is the `o` keybinding in evil-org-mode). Not sure it's worth looking at beyond seeing if they're just calling an org-mode function.

Reply:
`habitica--insert-task` function will figure out how to insert the task, the problem is where in the buffer to insert it relative to the other tasks, and possibly respecting the ordering of tasks in the habitica system.